### PR TITLE
Optimizing slow inference tests

### PR DESCRIPTION
**Pull request by sleepy-owl**
Related to #2324 

These optimizations reduce the running time of the tests, while still ensuring that they pass with high probability

- Updated the parameters of more slow tests
- Manually verified that the tests pass on GPU
- Also tried running multiple times with different seeds and verified that it passes with multiple seeds

Please let me know if this looks good or if there are any additional steps that I need to perform to verify these changes

### CODE REVIEW

**Comment by sleepy-owl** [comment]
CC @fritzo @martinjankowiak

File: tests/infer/test_elbo_mapdata.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def test_elbo_mapdata(batch_size, map_type):
     lam = torch.tensor([6.0, 4.0])
     data = []
     sum_data = torch.zeros(2)
+    params = {("iplate", 8): (100, 0.018), ("iplate", None): (100, 0.013),
```
Comment (martinjankowiak) [inline tests/infer/test_elbo_mapdata.py]: nit: can you replace these sorts of thing with `mark.parameterize' usage?

see e.g. [here](https://github.com/pyro-ppl/pyro/blob/1888efc05199bcb9319d5d4869d417988a4104d8/tests/contrib/autoguide/test_mean_field_entropy.py#L26)

Comment (sleepy-owl) [reply tests/infer/test_elbo_mapdata.py]: done

File: tests/infer/test_elbo_mapdata.py [hunk 2]
```diff
@@ -47,7 +50,7 @@ def add_data_point(x, y):
     analytic_log_sig_n = -0.5 * torch.log(analytic_lam_n)
     analytic_loc_n = sum_data * (lam / analytic_lam_n) +\
         loc0 * (lam0 / analytic_lam_n)
-    n_steps = 7000
+    n_steps = params[0]
```
Comment (martinjankowiak) [inline tests/infer/test_elbo_mapdata.py]: nit: instead of using `params[0]` can you please expand `params` to `n_steps,lr` or the like?

Comment (martinjankowiak) [reply tests/infer/test_elbo_mapdata.py]: i.e. directly in `mark.parametrize`

File: tests/infer/test_elbo_mapdata.py [hunk 3]
```diff
@@ -17,9 +17,12 @@
 
 @pytest.mark.stage("integration", "integration_batch_1")
 @pytest.mark.init(rng_seed=161)
-@pytest.mark.parametrize("batch_size", [3, 8, None])
-@pytest.mark.parametrize("map_type", ["plate", "iplate", "range"])
-def test_elbo_mapdata(batch_size, map_type):
+@pytest.mark.parametrize("map_type, batch_size, params", [("iplate", 3, (7000, 0.0008)), ("iplate", 8, (100, 0.018)),
```
Comment (fritzo) [inline tests/infer/test_elbo_mapdata.py]: I think what @martinjankowiak is suggesting is to write this as
@pytest.mark.parametrize("map_type, batch_size, n_steps, lr", [
    ("iplate", 3, 7000, 0.0008),
    ("iplate", 8, 100, 0.018),
    ...
])

File: tests/infer/test_elbo_mapdata.py (hunk 2 above)
Comment (sleepy-owl) [reply tests/infer/test_elbo_mapdata.py]: done

**Comment by sleepy-owl** [comment]
Hi @fritzo, seems like the build failed. But I dont see any relevant error message. Do you know why it failed?

**Pull request merged**